packages/core-data/src/fetch/index.js

```diff
@@ -1,2 +1,56 @@
-export { default as __experimentalFetchLinkSuggestions } from './__experimental-fetch-link-suggestions';
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '../';
+import { default as fetchLinkSuggestions } from './__experimental-fetch-link-suggestions';
+
+export const __experimentalFetchLinkSuggestions = fetchLinkSuggestions;
 export { default as __experimentalFetchUrlData } from './__experimental-fetch-url-data';
+
+export function __experimentalUseLinkControlEntitySearch() {
+	const settings = useSelect(
+		( select ) => select( blockEditorStore ).getSettings(),
+		[]
+	);
+	// The function should either be undefined or a stable function reference
+	// throughout the editor lifetime, much like importing a function from a
+	// module.
+	const { pageOnFront, pageForPosts } = useSelect( ( select ) => {
+		const { canUser, getEntityRecord } = select( coreStore );
+
+		const siteSettings = canUser( 'read', 'settings' )
+			? getEntityRecord( 'root', 'site' )
+			: undefined;
+
+		return {
+			pageOnFront: siteSettings?.page_on_front,
+			pageForPosts: siteSettings?.page_for_posts,
+		};
+	}, [] );
+
+	return useCallback(
+		async ( val, suggestionsQuery ) => {
+			return (
+				await fetchLinkSuggestions( val, suggestionsQuery, settings )
```

Comment:
Could we instead access the original `settings.__experimentalFetchLinkSuggestions` and call that if it's defined? That might provide backwards compatibility we need.